### PR TITLE
[VarDumper] Add maxDepth & maxStringLength display options

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q | A |
| --- | --- |
| Branch? | master |
| Bug fix? | no |
| New feature? | yes |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | #16741 |
| License | MIT |
| Doc PR | - |

Takes over #18148 to add display options to html dumps.
Status: needs work

### CODE REVIEW

File: src/Symfony/Component/VarDumper/Dumper/HtmlDumper.php [hunk 1]
```diff
@@ -43,7 +43,15 @@ class HtmlDumper extends CliDumper
         'meta' => 'color:#B729D9',
         'key' => 'color:#56DB3A',
         'index' => 'color:#1299DA',
+        'str .max-string-length b' => 'color:#A0A0A0',
     );
+    protected $displayOptions = array(
```
Comment (stof) [inline src/Symfony/Component/VarDumper/Dumper/HtmlDumper.php]: should be private

File: src/Symfony/Component/VarDumper/Dumper/HtmlDumper.php [hunk 2]
```diff
@@ -324,6 +408,15 @@ function isCtrlKey(e) {
     border: 0;
     outline: none;
 }
+pre.sf-dump .max-string-length.expanded .collapsed {
+    display:none;
+}
+pre.sf-dump .max-string-length.collapsed .expanded {
+    display:none
```
Comment (phansys) [inline src/Symfony/Component/VarDumper/Dumper/HtmlDumper.php]: Missing delimiter.

File: src/Symfony/Component/VarDumper/Dumper/HtmlDumper.php [hunk 3]
```diff
@@ -117,7 +137,8 @@ protected function getDumpHeader()
             return $this->dumpHeader;
         }
 
-        $line = <<<'EOHTML'
+        $line = json_encode($this->displayOptions, JSON_FORCE_OBJECT);
+        $line = <<<EOHTML
```
Comment (xabbuh) [inline src/Symfony/Component/VarDumper/Dumper/HtmlDumper.php]: We have to stick with nowdoc here or alternatively have to escape `$` characters below.

**Comment by fabpot** [comment]
@nicolas-grekas  Can you finish this one?

**Comment by fabpot** [comment]
@nicolas-grekas What's the status of this one?

**Comment by nicolas-grekas** [comment]
I'm on it :)

**Comment by nicolas-grekas** [comment]
I finally managed to finish this one.

@MGDSoft FYI, I removed the initDepth/maxDepth distinction: it felt strange when playing with the UI. I kept only your initDepth behavior, renamed as maxDepth.

See this fiddle to try the resulting behavior (maxDepth=2):
https://jsfiddle.net/Lz08fgct/

Status: needs review

File: src/Symfony/Component/VarDumper/Dumper/HtmlDumper.php [hunk 4]
```diff
@@ -76,6 +82,17 @@ public function setStyles(array $styles)
     }
 
     /**
+     * Configures display options.
+     *
+     * @param array $displayOptions A map of display options to customize the behavior.
```
Comment (fabpot) [inline src/Symfony/Component/VarDumper/Dumper/HtmlDumper.php]: no dot at the end

**Comment by fabpot** [comment]
Thank you @nicolas-grekas.

File: src/Symfony/Component/VarDumper/Dumper/HtmlDumper.php (hunk 4 above)
Comment (fabpot) [inline src/Symfony/Component/VarDumper/Dumper/HtmlDumper.php]: ahah, you've been doing that for a long time, working on a PR to fix that :)

**Comment by HeahDude** [comment]
Nice :)

**Pull request merged**